include/tvm/runtime/data_type.h

```diff
@@ -211,6 +215,13 @@ class DataType {
   static DataType Handle(int bits = 64, int lanes = 1) {
     return DataType(kHandle, bits, lanes);
   }
+  /*!
+   * \brief Construct a None type.
+   * \return The constructed data type.
+   */
+  static DataType None() {
```

Comment:
to be consistent with the VoidType, perhaps we can use Void here? See also https://github.com/apache/incubator-tvm/blob/master/include/tvm/ir/type.h#L375

Also need to update GetRuntimeType and GetType here https://github.com/apache/incubator-tvm/blob/master/include/tvm/tir/op.h#L60

Reply:
I'm not sure what to do in `GetType`.  It checks for a pointer type, and for other expressions it just returns the DataType member, so it seems like there is no extra handling needed.

Btw, the if statement doesn't seem to be doing anything, since it's followed by the exact same return statement:
https://github.com/apache/incubator-tvm/blob/master/src/tir/ir/op.cc#L61

Reply:
It would be great to have a single VoidType, so GetType(DataType::Void()) better returns VoidType() and GetRuntimeType(VoidType()) returns the DataType variant.

The main reason of the additional return is that we suppose to add an WARNING if the check does not pass, see also the relations between runtime type and type in https://github.com/apache/incubator-tvm/blob/master/include/tvm/ir/type.h#L29

Reply:
Ah, of course.  I didn't notice that `GetType` returned `Type`.